### PR TITLE
Add snow troubleshooting guide

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

[#1111](https://github.com/aws/eks-anywhere/issues/1111)

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 1]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+        * If there are more than 1 control plan machine provisioned on the same device and there are devices with no control plane machine, you need to rebalance control plane nodes. Follow the [rebalance nodes]({{< relref "./troubleshooting/#how-to-rebalance-nodes" >}}) procedure to rebalance your control plane nodes in order to recover high availability.
+        * If you have your control plane nodes evenly distributed with 1 device having at most 1 control plane node, you are done with the recovery.
+
+#### How to rebalance nodes
+
+1. Confirm the machines you want to delete and get their node name from the NODENAME column.
+
+    You can determine which machines need to be deleted by referring to the AGE column. The newly-generated machines have short AGE. Delete those new etcd/control plane machine nodes which are not the only etcd/control plane machine nodes on their devices.
+
+1. Cordon nodes so no further workloads are scheduled to run on it.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:611]:
```suggestion
1. Cordon each node so no further workloads are scheduled to run on it.
```

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 2]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+
+### Device replacement
+
+There might be some reasons which can require device replacement:
+
+* When a subset of devices are determined to be broken and you want to join a new device into current cluster.
+* When a subset of devices are offline and come back with a new device IP.
+
+**To upgrade a cluster with new devices**:
+1. Add new certificate to certificate file and new credentials to credential file.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:639]:
```suggestion
1. Add new certificates to the certificate file and new credentials to the credential file.
```

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 3]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+
+1. Change the device list in your cluster yaml configuration file and use the eksctl anywhere upgrade cluster command.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:641]:
```suggestion
1. Change the device list in your cluster yaml configuration file and use the `eksctl anywhere upgrade cluster` command.
```

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 4]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+
+### Node outage
+
+**Unintentional instance outage**
+
+When an instance is in exception status (for example, terminated/stopped for some reason), it will be discovered automatically by Amazon EKS Anywhere and there will be a new replacement instance node created after 5 minutes. The new node will be provisioned to devices based on even provision strategy. In this case, the new node will be provisioned to a device with the fewest number of machines of the same type. Sometimes, more than one device will have the same number of machines of this type. Thus, we cannot guarantee it will be provisioned on the original device.
+
+**Intentional node replacement**
+
+If you want to replace an unhealthy node which didn't get detected by Amazon EKS Anywhere automatically, you can follow the these steps.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:655]:
```suggestion
If you want to replace an unhealthy node which didn't get detected by Amazon EKS Anywhere automatically, you can follow these steps.
```

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 5]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+
+>**_NOTE:_** do not delete all worker machine nodes or control plane nodes or etcd nodes at the same time. Make sure you delete machine nodes one by one.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:657]:
```suggestion
>**_NOTE:_** Do not delete all worker machine nodes or control plane nodes or etcd nodes at the same time. Make sure you delete machine nodes one by one.
```

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md [hunk 6]
```diff
@@ -534,3 +535,170 @@ $ kubectl get deployments -n eksa-system
...
+
+If your Amazon EKS Anywhere cluster creation failed and the `eksctl anywhere delete cluster -f eksa-cluster.yaml` command cannot be run successfully, manually delete a few resources before trying the command again. Run the following commands from the computer on which you set up the AWS configuration and have the Snowball Edge Client installed. If you are using multiple Snowball Edge devices, run these commands on each.
+
+```sh
+// get the list of instance ids that are created for Amazon EKS Anywhere cluster, 
+// that can be identified by cluster name in the tag of the output
+$ aws ec2 describe-instances --endpoint http://snowball-ip:8008 --profile profile-name
+
+// the next two commands are for deleting DNI, this needs to be done before deleting instance
+$ PATH_TO_Snowball_Edge_CLIENT/bin/snowballEdge describe-direct-network-interfaces --endpoint https://snowball-ip --manifest-file path-to-manifest-file --unlock-code unlock-code
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:693]: I'm just curious what we have this "PATH_TO_Snowball_Edge_CLIENT" here. Shouldn't it just be put in the user's $PATH, so they could just type "snowballEdge"?

**Review by chrisnegus — COMMENTED** [review]
This content is really great! I just have some small wording suggestions and a question.

File: docs/content/en/docs/tasks/troubleshoot/troubleshooting.md (hunk 6 above)
Comment (jiayiwang7) [reply docs/content/en/docs/tasks/troubleshoot/troubleshooting.md:694]: added https://github.com/aws/eks-anywhere/pull/4776/files#diff-d6526994e6cc4a190722462afeac372b842fb107df0e61cedd96788b1878ee3fR552 and https://github.com/aws/eks-anywhere/pull/4776/files#diff-d6526994e6cc4a190722462afeac372b842fb107df0e61cedd96788b1878ee3fR686 for instruction on installing snowball edge client

**Pull request merged**